service/src/main/java/datawave/microservice/query/mapreduce/MapReduceQueryJobRunner.java

```diff
@@ -0,0 +1,272 @@
+    }
+    
+    private void submit(MapReduceQueryStatus mapReduceQueryStatus) {
+        try {
+            // submit the job
+            // @formatter:off
+            StringBuilder name = new StringBuilder()
+                    .append(mapReduceQueryStatus.getJobName())
+                    .append("_sid_")
+                    .append(ProxiedEntityUtils.getShortName(mapReduceQueryStatus.getCurrentUser().getPrimaryUser().getName()))
+                    .append("_id_")
+                    .append(mapReduceQueryStatus.getId());
+            // @formatter:on
+            
+            MapReduceJob mapReduceJob = mapReduceJobs.get(mapReduceQueryStatus.getJobName()).get();
+            
+            Job job;
+            try {
+                job = Job.getInstance(configuration, name.toString());
+                mapReduceJob.initializeConfiguration(queryLogicFactory, accumuloConnectionFactory, mapReduceQueryStatus, job);
+            } catch (Exception e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.LOGIC_CONFIGURATION_ERROR, e);
+                log.error(qe.getMessage(), e);
+                throw qe;
+            }
+            
+            // Enforce that certain InputFormat classes are being used here.
+            if (mapReduceQueryProperties.isRestrictInputFormats()) {
+                // Make sure that the job input format is in the list
+                Class<? extends InputFormat<?,?>> ifClass;
+                try {
+                    ifClass = job.getInputFormatClass();
+                } catch (ClassNotFoundException e) {
+                    QueryException qe = new QueryException(DatawaveErrorCode.INPUT_FORMAT_CLASS_ERROR, e);
+                    log.error(qe.getMessage(), e);
+                    throw qe;
+                }
+                if (!mapReduceQueryProperties.getValidInputFormats().contains(ifClass)) {
+                    IllegalArgumentException e = new IllegalArgumentException(
+                                    "Invalid input format class specified. Must use one of " + mapReduceQueryProperties.getValidInputFormats());
+                    QueryException qe = new QueryException(DatawaveErrorCode.INVALID_FORMAT, e);
+                    log.error(qe.getMessage(), e);
+                    throw qe;
+                }
+            }
+            
+            try {
+                job.submit();
+            } catch (Exception e) {
+                QueryException qe = new QueryException(DatawaveErrorCode.MAPREDUCE_JOB_START_ERROR, e);
+                log.error(qe.getMessage(), qe);
+                throw qe;
+            }
+            
+            JobID mapReduceJobId = job.getJobID();
+            log.info("JOB ID: " + mapReduceJobId);
+            
+            // This is used to lookup the query id when the mapreduce callback comes in
+            // TODO: JWO: Should we use a CustomJobEndNotifier that passes the map reduce query id to the update state call instead?
```

Comment:
todo

Reply:
resolved all of the todos.  some got fixed, some don't apply anymore, and some were made into issues.